appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -197,12 +201,26 @@ internal class ExperienceRenderer(
         stateMachines.getOwner(renderContext)?.stateMachine?.handleAction(StartStep(stepReference))
     }
 
-    suspend fun preview(experienceId: String): Boolean {
-        repository.getExperiencePreview(experienceId)?.let {
-            return show(it)
-        }
+    sealed class PreviewResponse {
+        data class Success(val experience: Experience) : PreviewResponse()
+        data class NetworkError(val throwable: Throwable?) : PreviewResponse()
+        data class HttpError(val code: Int?) : PreviewResponse()
+        data class RenderDeferred(val frameId: String, val name: String) : PreviewResponse()
+    }
 
-        return false
+    suspend fun preview(experienceId: String): PreviewResponse {
+        repository.getExperiencePreview(experienceId).run {
```

Comment:
Then, you could do the mapping to `PreviewResponse` here, since it makes sense that the renderer will need to return that detail back to the caller for the error messages